File: RouterLib/src/main/java/com/kronos/router/model/RouterOptions.java
package com.kronos.router.model;

import android.app.Activity;
import android.os.Bundle;

import com.kronos.router.RouterCallback;

/**
 * Created by zhangyang on 16/7/16.
 */
public class RouterOptions {
    Class<? extends Activity> _activityClass;
    RouterCallback _callback;
    Bundle _defaultParams;

    public RouterOptions() {
        if (_defaultParams == null) {
            _defaultParams = new Bundle();
        }
    }

    public RouterOptions(Class<? extends Activity> klass) {
        this.setOpenClass(klass);
    }

    public RouterOptions(Bundle defaultParams) {
        this.setDefaultParams(defaultParams);
    }

    public RouterOptions(Bundle defaultParams, Class<? extends Activity> klass) {
        this.setDefaultParams(defaultParams);
        this.setOpenClass(klass);
    }

    public void setOpenClass(Class<? extends Activity> klass) {
        this._activityClass = klass;
    }

    public Class<? extends Activity> getOpenClass() {
        return this._activityClass;
    }

    public RouterCallback getCallback() {
        return this._callback;
    }

    public void setCallback(RouterCallback callback) {
        this._callback = callback;
    }

    public void setDefaultParams(Bundle defaultParams) {
        if (_defaultParams == null) {
            _defaultParams = new Bundle();
        }
        if (defaultParams != null) {
            _defaultParams.putAll(defaultParams);
        }
    }


    public void putParams(String key, String value) {
        _defaultParams.putString(key, value);
    }

    public Bundle getDefaultParams() {
        return this._defaultParams;
    }
}

File: RouterLib/src/main/java/com/kronos/router/model/RouterOptions.kt
package com.kronos.router.model

import android.app.Activity
import android.os.Bundle

import com.kronos.router.RouterCallback
import com.kronos.router.interceptor.Interceptor

/**
 * Created by zhangyang on 16/7/16.
 */
class RouterOptions {

    var openClass: Class<out Activity>? = null
    var callback: RouterCallback? = null
    private val _defaultParams: Bundle by lazy {
        Bundle()
    }
    var weight = 0
    val interceptors by lazy {
        mutableListOf<Interceptor>()
    }

    var defaultParams: Bundle?
        get() = this._defaultParams
        set(defaultParams) {
            _defaultParams.putAll(defaultParams)
        }

    constructor() {

    }

    constructor(defaultParams: Bundle?) {
        this.defaultParams = defaultParams
    }


    fun putParams(key: String, value: String) {
        _defaultParams.putString(key, value)
    }

    fun addInterceptor(interceptor: Interceptor) {
        interceptors.add(interceptor)
    }

    fun addInterceptors(interceptor: List<Interceptor>) {
        interceptors.addAll(interceptor)
    }
}